nemo_curator/utils/distributed_utils.py

```diff
@@ -261,6 +261,20 @@ def _set_torch_to_use_rmm():
     torch.cuda.memory.change_current_allocator(rmm_torch_allocator)
 
 
+def select_and_sort_columns(
+    df: Union[dd.DataFrame, dask_cudf.DataFrame],
+    columns: List[str],
+    add_filename: bool,
+) -> Union[dd.DataFrame, dask_cudf.DataFrame]:
+    # TODO : Reviewer TAL if filetype check is needed
```

Comment:
Originally I added the "json" and "jsonl" check because `columns` is already added as a kwarg to `read_parquet`, so I didn't want to filter twice in the Parquet case.

Reply:
That makes sense. It was originally missing `pickle` but I didn't realize about column selection already for `parquet`. I'll exclude `parquet` in that case. WDYT?

Reply:
IIRC `pickle` does not go through this file path, only JSON and Parquet.

Reply:
Oh I just noticed that `read_pandas_pickle` already handles the `column` case, but small nit that it doesn't sort the columns the same way as in for json / parquet path.

Reply:
Lmk if we need the sorting behavior, if so we'll need a variation of `select_and_sort_columns` to be called after our `read_*` method

Reply:
IMO we don't have to worry about the Pickle case, thanks.